packages/react-admin-core/src/EditDialog.tsx

```diff
@@ -84,7 +96,9 @@ const EditDialogInner: React.FunctionComponent<IProps & IHookProps> = ({ selecti
                             <DirtyHandlerApiContext.Consumer>
                                 {injectedDirtyHandlerApi => {
                                     dirtyHandlerApi = injectedDirtyHandlerApi; // TODO replace by ref on <DirtyHandler>
-                                    return (
+                                    return renderSaveButton ? (
+                                        React.cloneElement(renderSaveButton, { onClick: handleSaveClick })
```

Comment:
cloneElement?

why not call `renderSavebutton()`

Reply:
I changed renderSaveButton to be a function instead of a react component. This makes way more sense.
Is that what you had in mind?